lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -0,0 +1,249 @@
+    HTTP_422_UNPROCESSABLE_ENTITY,
+)
+
+from openedx.core.djangoapps.catalog.utils import get_programs
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+from lms.djangoapps.program_enrollments.api.v1.constants import CourseEnrollmentResponseStatuses
+from lms.djangoapps.program_enrollments.models import ProgramEnrollment, ProgramCourseEnrollment
+from lms.djangoapps.program_enrollments.serializers import ProgramCourseEnrollmentRequestSerializer
+
+
+class ProgramEnrollmentsView(APIView):
+    """
+    POST view for ProgramEnrollments
+    """
+
+    def get(self, request, *args, **kwargs):
+        return HttpResponse('result')
+
+
+class ProgramSpecificViewMixin(object):
+    """
+    A mixin for views that operate on or within a specific program.
+    """
+    def __init__(self, *args, **kwargs):
+        super(ProgramSpecificViewMixin, self).__init__(*args, **kwargs)
+        self._program = None
+
+    @property
+    def program(self):
+        """
+        The program specified by the `program_uuid` URL parameter.
+        """
+        if self._program is None:
+            program = get_programs(uuid=self.kwargs['program_uuid'])
+            if program is None:
+                raise Http404()
+            self._program = program
+        return self._program
+
+
+class ProgramCourseRunSpecificViewMixin(ProgramSpecificViewMixin):
+    """
+    A mixin for views that operate on or within a specific course run in a program
+    """
+    def __init__(self, *args, **kwargs):
+        super(ProgramCourseRunSpecificViewMixin, self).__init__(*args, **kwargs)
+        self._course_run = None
+        self._course_key = None
+
+    def check_existence_and_membership(self):
+        """
+        Attempting to look up the course and program will trigger 404 responses if:
+        - The program does not exist
+        - The course run (course_key) does not exist
+        - The course run is not part of the program
+        """
+        self._parse_run_and_key()
+
+    @property
+    def course_run(self):
+        """
+        The course run specified by the `course_id` URL parameter.
+        """
+        if self._course_run is None:
+            self._parse_run_and_key()
+        return self._course_run
+
+    @property
+    def course_key(self):
+        """
+        The course key for the course run specified by the `course_id` URL parameter.
+        """
+        if self._course_key is None:
+            self._parse_run_and_key()
+        return self._course_key
+
+    def _parse_run_and_key(self):
+        """
+        Parse the course_run and course_key fields from the course_id url parameter
+
+        Raises Http404 if the program or course_run does not exist or if the course_run
+        is not in the program
+        """
+        try:
+            course_key = CourseKey.from_string(self.kwargs['course_id'])
+            course = CourseOverview.get_from_id(course_key)
```

Comment:
Consider not assigning this to `course` - you could assign to `_` or not assign at all. I got confused when reading the rest of the function because there's another `course` variable.